sycl/test-e2e/KernelAndProgram/free_function_kernels.cpp

```diff
@@ -0,0 +1,274 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// UNSUPPORTED: hip
```

Comment:
What makes hip special here?

Reply:
The proper check is availability of usm_shared memory which the test uses. I've corrected the test.